datamol/data/salts.smi

```diff
@@ -0,0 +1,233 @@
+OC(=O)C=CC(=O)O	Maleate and Fumarate
+CS(=O)(=O)O	Mesylate
+OP(=O)=O	Metaphosphate
+COS(=O)(=O)O	Methosulfate
+[Mg]	Magnesium
+OP(=O)(O)F	Monofluorophosphate
+[Na]	Sodium
+OS(=O)(=O)c1cccc2c(cccc12)S(=O)(=O)O	Napadisilate
+OS(=O)(=O)c1ccc2ccccc2c1	Napsylate
+O[N](=O)O	Nitrate
+OC(=O)C(=O)O	Oxalate
+CCCCCCCCCCCCCCCC(=O)O	Palmitate
+OC(=O)c1cc2ccccc2c(Cc3c(O)c(cc4ccccc34)C(=O)O)c1O	Pamoate
+OCl(=O)(=O)=O	Perchlorate
+Nc1ccc(cc1)P(=O)(O)O	Phosphanilate
+OP(=O)(O)O	Phosphate
+Oc1c(cc(cc1[N+](=O)[O-])[N+](=O)[O-])[N+](=O)[O-]	Picrate
+C1CNCCN1	Piperazine
+CC(O)CO	Propylene Glycol
+O=C1NS(=O)(=O)c2ccccc12	Saccharin
+OC(=O)c1ccccc1O	Salicylate
+[Ag]	Silver
+[Sr]	Strontium
+OC(=O)CCC(=O)O	Succinate
+OS(=O)(=O)O	Sulfate
+OC(=O)c1cccc(c1O)S(=O)(=O)O	Sulfosalicylate
+[S-2]	Sulphide
+OC(=O)c1ccc(cc1)C(=O)O	Terephthalate
+Cc1ccc(cc1)S(=O)(=O)O	Tosylate
+Oc1cc(Cl)c(Cl)cc1Cl	Triclofenate
+CCN(CC)CC	Triethylamine
+OC(=O)C(c1ccccc1)(c2ccccc2)c3ccccc3	Trifenatate
+OC(=O)C(F)(F)F	Triflutate
+NC(CO)(CO)CO	Tromethamine
+CCCCC1CCC(CC1)C(=O)O	Buciclate
+CCCC(=O)O	Butyrate
+CCCCCC(=O)O	Caproate
+CC12CCC(CC1)(C=C2)C(=O)O	Cyclotate
+OC(=O)CCC1CCCC1	Cypionate
+CN(C)CCC(=O)O	Daproate
+OC(=O)CN(CCN(CC(=O)O)CC(=O)O)CC(=O)O	EDTA
+CCCCCCCCC=CCCCCCCCC(=O)O	Elaidate and oleate
+CCCCCCC(=O)O	Enanthate
+CCOC(=O)O	Etabonate
+COCCO	Ethanediol
+OC(=O)CNC(=O)c1ccccc1	Etiprate
+CCC(CC)C(=O)OCO	Etzadroxil
+CCCCCCCCCCCCCCOP(=O)(O)O	Fostedate
+OC(=O)c1occc1	Furoate
+OC(=O)c1ccccc1C(=O)c2ccc(O)cc2	Hybenzate
+CCCCCCCCCCCC(=O)O	Laurate
+CC=C(C)C(=O)O	Mebutate
+COC(=O)CC(O)(CCCC(C)(C)O)C(=O)O	Mepesuccinate
+OC(=O)c1cccc(c1)S(=O)(=O)O	Metazoate
+CSCCC(N)C(=O)C	Methionil
+OC(=O)c1cccnc1	Nicotinate
+OO	Peroxide
+OC(=O)CCc1ccccc1	Phenpropionate
+OC(=O)Cc1ccccc1	Phenylacetate
+CC(C)(C)C(=O)O	Pivalate
+CCC(=O)O	Propionate
+CC(C)(C)CC(=O)O	Tebutate
+OCCN(CCO)CCO	Trolamine
+CCCCCCCCCCC(=O)O	Undecylate
+OC(=O)CCCCCCCCC=C	Undecylenate
+CCCCC(=O)O	Valerate
+O	Water
+OC(=O)c1ccc2ccccc2c1O	Xinafoate
+[Zn]	Zinc
+c1c[nH]cn1	Imidazole
+OCCN1CCOCC1	4-(2-Hydroxyethyl)morpholine
+CC(=O)Nc1ccc(cc1)C(=O)O	4-Acetamidobenzoic acid
+CC1(C)C(CCC1(C)C(=O)O)C(=O)O	Camphoric acid
+CCCCCCCCCC(=O)O	Capric acid
+CCCCCCCC(=O)O	Caprylic acid
+OC(=O)C=Cc1ccccc1	Cinnamic acid
+OC(C(O)C(O)C(=O)O)C(O)C(=O)O	Mucic acid
+OC(=O)c1cc(O)ccc1O	Gentisic acid
+OC(=O)CCCC(=O)O	Glutaric acid
+OC(=O)CCC(=O)C(=O)O	2-Oxoglutaric acid
+OCC(=O)O	Glycolic acid
+CC(C)C(=O)O	Isobutyric acid
+OC(C(=O)O)c1ccccc1	Mandelic acid
+OC(=O)c1cc(=O)nc(=O)n1	Orotic acid
+OC(=O)C1CCC(=O)N1	Pyroglutamic acid
+OC(C(O)C(=O)O)C(=O)O	Tartrate
+SC#N	Thiocyanic acid
+CI	Methyl Iodide
+OS(=O)O	Sulfurous Acid
+C1CCC(CC1)NC2CCCCC2	Dicyclohexylamine
+OS(=O)(=O)C(F)(F)F	Triflate
+Cc1cc(C)c(c(C)c1)S(=O)(=O)O	Mesitylene sulfonate
+OC(=O)CC(=O)O	Malonic acid
+OS(=O)(=O)F	Fluorosulfuric acid
+CC(=O)OS(=O)(=O)O	Acetylsulfate
+[H]	Proton
+[Rb]	Rubidium
+[Cs]	Cesium
+[Fr]	Francium
+[Be]	Beryllium
+[Ra]	Radium
+C(=O)C(O)C(O)C(O)C(O)C(=O)O	Glucuronate open form
```

Comment:
Do you confirm those two files are only available in `ChEMBL_Structure_Pipeline` and not RDKit? Cuz if they are available in RDKit we could just use them.

Reply:
I see a lot of solvents here. Is the `solvents.smi` file required then ?

Reply:
@hadim I have merged the predefined salts data from both rdkit and chembl structure pipeline. The later one covers more salt data. 
@maclandrol You are right. We can merge them and remove all salts and solvents at once (which will be probably the case in practice).

Reply:
> @maclandrol You are right. We can merge them and remove all salts and solvents at once (which will be probably the case in practice).

But wouldn't that be nice to leave the freedom to remove either only slat or solvent? and maybe have a new function `remove_salt_and_solvent()`

Reply:
The RDKit's salt file definition contains both (https://github.com/rdkit/rdkit/blob/master/Data/Salts.txt). Also it's potentially does not make sense to remove just the salt, but not the liquid medium if there is any.

Reply:
I think it could make sense as the solvent can sometime be modelled implicitly. But it's likely an edge case. So for completeness I think it's fine to propose both functions.

If you feel strongly otherwise, then all good for me.

Reply:
Maybe we could have a single `remove_salt_and_solvent` with two flags turned `true` by default `remove_solvent=true` and `remove_salt=true`?

Reply:
I am more for having something where the user can provide their own file definition (or fragment list) if they feel like they don't want to remove some subset, + potentially making this a function call in `dm.standardize_mol` (would simplify API in molfeat too).  

One reason is that the logic is the same for solvent and salt and on the other end, there are likely a lot of case by case on removing known solvents vs all liquids medium.

Reply:
Ok I am good having this kind of flexibility. But I feel like it's important to also have a default salt and/or solvent remover explicit function so the user does not have to think about file definition, fragments, etc but simply removing salt/solvent from a good enough defaut list.

Reply:
Yes, we should have either the file or list of smarts/smiles optional. I have some doubt about the salt/solvent file definition used here. Ping @Fransu86 for review.